packages/user/Tokens/plugin/src/lib.rs

```diff
@@ -173,17 +173,23 @@ impl Helpers for TokensPlugin {
 
 impl User for TokensPlugin {
     fn credit(token_id: u32, debitor: String, amount: Decimal, memo: String) -> Result<(), Error> {
-        assert_authorized_with_whitelist(FunctionName::credit, vec!["homepage".into()])?;
+        let amount: Quantity = Self::decimal_to_u64(token_id, amount)?.into();
+        
+        if amount.value > 0 {
+            assert_authorized_with_whitelist(FunctionName::credit, vec!["homepage".into()])?;
+            
+            let packed_args = Actions::credit {
+                amount,
+                memo: memo.try_into().unwrap(),
+                debitor: debitor.as_str().into(),
+                token_id,
+            }
+            .packed();
 
-        let packed_args = Actions::credit {
-            amount: Self::decimal_to_u64(token_id, amount)?.into(),
-            memo: memo.try_into().unwrap(),
-            debitor: debitor.as_str().into(),
-            token_id,
+            add_action_to_transaction(Actions::credit::ACTION_NAME, &packed_args)
+        } else {
+            Ok(())
```

Comment:
I don't think this is an improvement. The caller *should* be forced to acknowledge that it's not doing anything, a silent no-op.

Proposed alternatives that let the caller decide if a no-op is the right behavior:
* Leave as is, let the caller check if the amount > 0
* Return an error, let the caller handle the error

Reply:
To use explicit error code